File: backend/src/otter_password_manager/domain/entities/password_entry.py
from dataclasses import dataclass
from datetime import datetime


@dataclass(frozen=True, slots=True)
class PasswordEntry:
    id: int
    owner_id: int
    service_name: str
    username: str
    encrypted_password: str
    website: str | None
    notes: str | None
    created_at: datetime
    updated_at: datetime

File: Assets/OtterPasswordManager/Runtime/Application/Models/PasswordEntry.cs
using System;
using Newtonsoft.Json;

namespace OtterPasswordManager.Application.Models
{
    public sealed class PasswordEntry
    {
        [JsonProperty("id")]
        public int Id { get; set; }

        [JsonProperty("service_name")]
        public string ServiceName { get; set; }

        [JsonProperty("username")]
        public string Username { get; set; }

        [JsonProperty("password")]
        public string Password { get; set; }

        [JsonProperty("website")]
        public string Website { get; set; }

        [JsonProperty("notes")]
        public string Notes { get; set; }

        [JsonProperty("created_at")]
        public DateTimeOffset CreatedAt { get; set; }

        [JsonProperty("updated_at")]
        public DateTimeOffset UpdatedAt { get; set; }
    }
}
